index.js

```diff
@@ -0,0 +1,213 @@
+// 1
+function getDeltaFunction(deltaObject) {
+  return function(subDelta = {}) {
+    const delta = {...deltaObject, ...subDelta}
+    return  delta.b * delta.b - 4 * delta.a * delta.c;
+  }
+}
+
+// delta = b*b - 4*a*c
+const getDelta = getDeltaFunction({ a: 1, b: 2, c: 3 });
+console.log(getDelta()); // -8
+console.log(getDelta({ b: 12 })); // 132
+console.log(getDelta({ a: 4, b: 10 })); // 52
+console.log(getDelta({ a: 4, b: 0 })); // -48
+
+// The nested function returns NaN if any of the arguments are missing
+console.log(getDeltaFunction({ a: 1, c: 3 })()); // NaN
+console.log(getDeltaFunction({ a: 1, c: 3 })({ b: 15 })); // 213
+
+// 2
+function positiveSum(numbers) {
+  let numbersSum = 0;
+
+  numbers.forEach(function(number) {
+    if (number > 0) {
+      numbersSum = numbersSum + number;
+    }
+  })
+
+  return numbersSum;
+}
+
+console.log(positiveSum([1, 5, 10]));
+console.log(positiveSum([-1, 5, -10]));
+
+// 3
+function countSheep(shed) {
+  return shed.filter(Boolean).length;
+}
+
+const shed = [true, false, true, false, true];
+console.log(countSheep(shed)); //3
+
+// 4
+function findNeedle(haystack) {
+  const needle = (el) => el === "needle";
```

Comment:
I told you not to use arrow functions and you applied this comment in another place of this PR. Did you write this code yourself, or did you use ChatGPT or copied the code from the internet? :thinking:

Reply:
Also, don't use abbreviations. Please don't make the same mistakes multiple times.

Reply:
sorry, don't know what arrow function is so didn't know that this was either - fixed